project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module

```diff
@@ -474,8 +491,77 @@ function c4m_user_og_c4m_cta_block_alter(&$block) {
   $variables = array(
     'button_label' => $types[$membership_request_value]['button_label'],
     'button_icon' => $types[$membership_request_value]['button_icon'],
-    'url' => url('/user'),
+    'url' => url('group/join/' . $group->nid),
```

Comment:
@soniCaH @janlemmens 
with the options I get the same result,
I don't understand your concern, can you please let me know what the url you would like to achieve here?
thanks.